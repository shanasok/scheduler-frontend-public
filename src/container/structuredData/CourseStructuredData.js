import React from 'react';

export function makeCourseSchema(course) {
    const courseSummary = (course) ? course.summary : "";
    const courseName = (course) ? course.name : "";

    return {
        '@context': 'http://schema.org',
        '@type': 'Course',
        'name': courseName,
        'description': courseSummary,
    }
}

// Structured Data element to improve SEO ranking
// https://developers.google.com/search/docs/data-types/course
export class CourseStructuredData extends React.Component {

    // constructor(props) {
    //     super(props);
    //
    //     this.makeCourseSchema = this.makeCourseSchema.bind(this);
    // }
    //
    //
    // makeCourseSchema(course) {
    //     const courseSummary = (course) ? course.summary : "";
    //     const courseName = (course) ? course.name : "";
    //
    //     return {
    //         '@context': 'http://schema.org',
    //         '@type': 'Course',
    //         'name': courseName,
    //         'description': courseSummary,
    //     }
    // }

    render() {
        const schema = JSON.stringify(makeCourseSchema(this.props.course));

        return <script type="application/ld+json">
            {schema}
        </script>;
    }
}

export default CourseStructuredData;