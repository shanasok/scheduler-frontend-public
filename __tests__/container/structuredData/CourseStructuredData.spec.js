import React from 'react';
import {CourseStructuredData, makeCourseSchema} from '../../../src/container/structuredData/CourseStructuredData';
import renderer from 'react-test-renderer';

it('renders correctly', () => {

    const course = {
        name: "testCourseName",
            summary: "testCourseSummary",
            subjects: [
            {
                name: "testSubjectName",
                summary: "testSubjectSummary",
                subjectNumber: 1
            }
        ]
    }

    const tree = renderer
        .create(<CourseStructuredData course={course}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('accesses course details correctly when course provided', () => {

    const course = {
        name: "testCourseName",
        summary: "testCourseSummary",
        subjects: [
            {
                name: "testSubjectName",
                summary: "testSubjectSummary",
                subjectNumber: 1
            }
        ]
    }

    const structuredDataSchema = makeCourseSchema(course);
    expect(structuredDataSchema).toBeDefined();
    expect(structuredDataSchema.name).toEqual(course.name);
    expect(structuredDataSchema.description).toEqual(course.summary);
});

it('accesses course details correctly when course not provided', () => {

    const structuredDataSchema = makeCourseSchema();
    expect(structuredDataSchema).toBeDefined();
    expect(structuredDataSchema.name).toEqual("");
    expect(structuredDataSchema.description).toEqual("");
});