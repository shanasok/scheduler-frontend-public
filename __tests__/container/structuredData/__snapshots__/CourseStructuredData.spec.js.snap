// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"http://schema.org","@type":"Course","name":"testCourseName","description":"testCourseSummary"}
</script>
`;
